core/classes/Core/User.php

```diff
@@ -157,32 +126,24 @@ public function find(string $value = null, string $field = 'id'): bool {
      *
      * @param int $group_id ID of group to give.
      * @param int $expire Expiry in epoch time. If not supplied, group will never expire.
-     * @param object|null $group_data Load data from existing query.
      *
      * @return bool True on success, false if they already have it.
      */
-    public function addGroup(int $group_id, int $expire = 0, $group_data = null): bool {
-        if (array_key_exists($group_id, $this->_groups)) {
+    public function addGroup(int $group_id, int $expire = 0): bool {
+        if (array_key_exists($group_id, $this->getGroups())) {
             return false;
         }
 
-        $this->_db->query(
-            'INSERT INTO `nl2_users_groups` (`user_id`, `group_id`, `received`, `expire`) VALUES (?, ?, ?, ?)',
-            [
-                $this->data()->id,
-                $group_id,
-                date('U'),
-                $expire
-            ]
-        );
-
-        if ($group_data === null) {
-            $group = Group::find($group_id);
-            if ($group) {
-                $this->_groups[$group_id] = $group;
-            }
-        } else {
-            $this->_groups[$group_id] = new Group($group_data);
+        $this->_db->query('INSERT INTO `nl2_users_groups` (`user_id`, `group_id`, `received`, `expire`) VALUES (?, ?, ?, ?)', [
+            $this->data()->id,
+            $group_id,
+            date('U'),
+            $expire
+        ]);
+
```

Comment:
Wouldn't it be better to check if the group exists before we add it into the database?

Reply:
it do check already in the code

Reply:
it do check already in the code